### PR TITLE
549 delete curve feature

### PR DESCRIPTION
**Pull request by SilkeSchomann**
Added a 'Delete Line' button in the settings dialog for 1D plots. Clicking it removes the LegendAndLineOptionsSetter widget for the respective line as well as the corresponding line in the cut plot.

**To test:**

1. Open a cut plot with several lines.
2. Open the settings dialog from the plot window.
3. Delete one of the lines.
4. Check that the corresponding settings widget is removed from the settings dialog.
5. Check that the corresponding line and its legend is removed from the plot.

Fixes #[549](https://github.com/mantidproject/mslice/issues/549).

### CODE REVIEW

File: mslice/plotting/plot_window/cut_plot.py [hunk 1]
```diff
@@ -217,6 +217,16 @@ def set_line_options_by_index(self, line_index, line_options):
 
         self._lines_visible[line_index] = line_options['shown']
 
+    def remove_line_by_index(self, line_index):
+        container = self._canvas.figure.gca().containers[line_index]
+        containers = self._canvas.figure.gca().containers
```
Comment (srikanthravipati) [inline mslice/plotting/plot_window/cut_plot.py:222]: We could change 
`container = self._canvas.figure.gca().containers[line_index]`
`containers = self._canvas.figure.gca().containers`
to 
`containers = self._canvas.figure.gca().containers `
`container = containers[line_index]`!

File: mslice/plotting/plot_window/cut_plot.py [hunk 2]
```diff
@@ -217,6 +217,16 @@ def set_line_options_by_index(self, line_index, line_options):
...
+        container[0].remove()
+        for line in container[1]:
+            line.remove()
+        for line in container[2]:
+            line.remove()
```
Comment (srikanthravipati) [inline mslice/plotting/plot_window/cut_plot.py:227]: Consider
`for i in range(2)`
      `for line in container[i+1]`
             `line.remove()`

File: mslice/plotting/plot_window/plot_options.py [hunk 3]
```diff
@@ -345,6 +352,14 @@ def __init__(self, line_options, color_validator):
         if self.color_validator is not None:
             self.line_color.currentIndexChanged.connect(lambda selected: self.color_valid(selected))
 
+        if line_options['shown'] is None or line_options['legend'] is None:
+            row5 = QtWidgets.QHBoxLayout()
+            layout.addLayout(row5)
```
Comment (srikanthravipati) [inline mslice/plotting/plot_window/plot_options.py:357]: I think these two lines could go after line 319 (after row3), as we always want row5 at least with delete button. Could they? 
If so, lines 341-342 and 355-357 could be deleted after they are added after line 319!

**Review by srikanthravipati — CHANGES_REQUESTED** [review]
The fix introduces delete functionality for each and every line in cut plots and works as intended.
I have made minor recommendations to consider. Overall, looks great :)

File: mslice/plotting/plot_window/plot_options.py (hunk 3 above)
Comment (SilkeSchomann) [reply mslice/plotting/plot_window/plot_options.py:357]: Yes, a line could be deleted even if it is not shown or does not have a legend.

**Review by srikanthravipati — APPROVED** [review]

**Pull request merged**